actions/swap-appsvc-slot/action.yml

```diff
@@ -0,0 +1,32 @@
+name: "Swap App Service Slot"
+description: "Incrementally shift traffic to staging slot, run canary script, shift traffic and finalize with slot swap."
+
+inputs:
+  resource_group_name:
+    description: "Azure Resource Group name"
+    required: true
+  web_app_name:
+    description: "Azure Web App name"
+    required: true
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: Check out repository
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      with:
+        fetch-depth: 0
+
+    - name: Set GitHub Path
+      shell: bash
+      env:
+        GITHUB_ACTION_PATH: ${{ github.action_path }}
+      run: echo "$GITHUB_ACTION_PATH" >> $GITHUB_PATH
+
+    - name: Incremental Rollout to Staging Slot
+      shell: bash
+      env:
+        RESOURCE_GROUP_NAME: ${{ inputs.resource_group_name }}
+        WEB_APP_NAME: ${{ inputs.web_app_name }}
+      run: rollout.sh "$RESOURCE_GROUP_NAME" "$WEB_APP_NAME"
```

Comment:
question: does this work even when calling the action from another repo?

Reply:
Yes! the previous step adds the "GITHUB_ACTION_PATH" to the PATH of the runner :)